docs/hoon/concepts.md

```diff
@@ -82,16 +77,53 @@ it's still fun and useful to [learn more](../../nock).
 
 ## Compilation concepts
 
+### <a name="hoon">`hoon`</a> (AST node)
+
+A [`hoon`](../reference) is the result of parsing a Hoon source 
+expression into an AST node. Because every Hoon programs is, 
+in its entirety, a single expression of Hoon, the result of parsing 
+a whole Hoon program into an AST is a single `hoon`.
+
+As the noun that the parser produces, a hoon is a tagged union of
+the form `[tag data]`, where the tag is a constant such as `%brts`,
+which is matched up with the appropriate type of data (often more 
+`hoon`s).  For example, the expression `:-(p q)`, once parsed into 
+an AST, becomes the tagged union:
+
+```
+[%clhp p=hoon q=hoon]
+```
+
+The `%clhp` is for "colhep". Keep in mind that `p` and `q` would be 
+parsed too.  So if `p` is 2 and `q` is 17, the parsed result is:
+
+```
+[%clhp p=[%sand p=%ud q=2] q=[%sand p=%ud q=17]]
```

Comment:
I think it's worth mentioning that `[%sand p=%ud q=2]` means a the type of a constant atom with value 2 and aura `@ud` (maybe we shouldn't mention auras yet). Could say something like:
```
Notice `p` and `q`, both of type `hoon`, have been parsed as `%sand`-tagged `hoon`s, which represent constant atoms.
```

Reply:
I thought about that.  Honestly though I'm wondering if I've already gone too far by mentioning `%clhp`.  It's a little tutorial-ish for a definitional doc.  (I assume we want a tutorial/spec distinction in our docs.)  I think it's worth bending the standards a little to make our definitions clearer.  Examples are often helpful.  But I don't want to get too far into the weeds, as long as the general idea is clearish.

That said, saying a little to explain `%sand` seems like a fair suggestion, and I'm not particularly opposed to it if everyone likes the idea.

Reply:
I made the change.